wire-ios-sync-engine/Source/SessionManager/UserSessionLoader.swift

```diff
@@ -129,10 +129,13 @@ final class UserSessionLoader {
             proxyCredentials: proxyCredentials
         )
 
-        let metadata: ResolvedBackendMetadata = if let newMetadata = newEnvironment?.metadata {
-            newMetadata
+        let metadata: ResolvedBackendMetadata
+        if let newMetadata = newEnvironment?.metadata {
+            metadata = newMetadata
+            // we store the metadata for NSE
+            try backendStore.storeBackendMetadata(metadata, for: accountID)
```

Comment:
suggestion: it may be nice to do this together with storing the new environment at the start of the method, then we can leave this part unchanged.

Reply:
done